pkg/v1/cli/pluginmanager/manager.go

```diff
@@ -425,6 +483,80 @@ func SyncPlugins(serverName string) error {
 	return nil
 }
 
+// InstallPluginsFromLocalSource installs plugin from local source directory
+func InstallPluginsFromLocalSource(pluginName, version, localPath string) error {
+	// Set default local plugin distro to localpath as while installing the plugin
+	// from local source we should take t
+	common.DefaultLocalPluginDistroDir = localPath
+
+	plugins, err := DiscoverPluginsFromLocalSource(localPath)
+	if err != nil {
+		return err
+	}
+
+	found := false
+
+	errList := make([]error, 0)
+	for idx := range plugins {
+		if pluginName == plugins[idx].Name || pluginName == cli.AllPlugins {
+			found = true
+			err := installOrUpgradePlugin("", &plugins[idx], plugins[idx].RecommendedVersion)
+			if err != nil {
+				errList = append(errList, err)
+			}
+		}
+	}
+	err = kerrors.NewAggregate(errList)
+	if err != nil {
+		return err
+	}
+	if !found {
+		return errors.Errorf("unable to find plugin '%v'", pluginName)
+	}
+	return nil
+}
+
+// DiscoverPluginsFromLocalSource returns the available plugins that are discovered from the provided local path
+func DiscoverPluginsFromLocalSource(localPath string) ([]plugin.Discovered, error) {
+	if localPath == "" {
+		return nil, nil
+	}
+
+	// Set default local plugin distro to localpath as while installing the plugin
+	// from local source we should take t
+	common.DefaultLocalPluginDistroDir = localPath
+
+	var pds []v1alpha1.PluginDiscovery
+
+	items, err := os.ReadDir(filepath.Join(localPath, "discovery"))
+	if err != nil {
+		return nil, errors.Wrapf(err, "error while reading local plugin manifest directory")
+	}
+	for _, item := range items {
+		if item.IsDir() {
+			pd := v1alpha1.PluginDiscovery{
+				Local: &v1alpha1.LocalDiscovery{
+					Name: "",
+					Path: filepath.Join(localPath, "discovery", item.Name()),
+				},
+			}
+			pds = append(pds, pd)
+		}
+	}
+
+	plugins, err := discoverPlugins(pds)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range plugins {
+		plugins[i].Scope = common.PluginScopeStandalone
+		plugins[i].Status = common.PluginStatusNotInstalled
+	}
```

Comment:
would it be meaningful to set the DiscoveryType here?

Reply:
Thanks. Yes, we should. Added it here.